src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationMockTests.cs

```diff
@@ -35,11 +35,10 @@ public class ConfigurationMockTests
 
         private static (ConfigurationClient service, TestPool<byte> pool) CreateTestService(MockHttpClientTransport transport)
         {
-            HttpPipelineOptions options = ConfigurationClient.CreateDefaultPipelineOptions();
+            var options = new HttpPipelineOptions();
             var testPool = new TestPool<byte>();
-            options.AddService(testPool, typeof(ArrayPool<byte>));
 
-            options.Transport = transport;
+            options.TransportPolicy = () => transport;
```

Comment:
I wanted to avoid having pipeline authors think about state management inside a policy that might be attached to multiple different pipelines. Makes things very simple, policy object might only be in one stop of a single pipeline.

Reply:
pipeline authors is much smaller subset of developers than those who will configure it. I think we should optimize for the people who configure the pipeline above people who will develop custom policies. Policies should be stateless and be reusable in pipelines.

Reply:
People who configure it would have nice extensions methods to use.

Reply:
> Policies should be stateless and be reusable in pipelines

Is there a reason for this?